project_2/src/link.cpp

```diff
@@ -0,0 +1,26 @@
+#include "link.hpp"
+#include "common_events.hpp"
+
+#include <vector>
+#include <cstdint>
+#include <queue>
+#include <memory>
+#include <unordered_map>
+
+Link::Link(std::shared_ptr<NetworkElement>& source, std::shared_ptr<NetworkElement>& destination, std::uint64_t speed, std::uint64_t last_process_time_ns):
+    source(source),
+    destination(destination),
+    speed(speed),
+    last_process_time_ns(last_process_time_ns)
+    {}
+
+std::shared_ptr<NetworkElement> Link::GetNextElement(std::uint64_t destination_id) {
+    return destination;
+}
+
+void Link::ReceivePacket(std::uint64_t current_time_ns, PacketHeader& packet, std::priority_queue<std::shared_ptr<Event>, std::vector<std::shared_ptr<Event>>, EventComparator>& all_events) {
+    std::uint64_t process_time = (packet.size + speed - 1) / speed;
```

Comment:
What is this formula? Specifically, where `+ speed - 1` does this formula come from?

Reply:
Also, do you take into account the propagation delay?

Reply:
Its just a way of performing division with rounding up to work with nanoseconds as an integer number
No, I'm not taking into account the propagation delay right now

Reply:
I think, I am confused. All the quantities in the brackets have different units (we add size, speed and a constant). Can you, please, elaborate on where the formula
```
process_time = (packet.size + speed - 1) / speed
```
comes from.